### DISCUSSION TITLE
Nginx-exporter docker version -  issue when access.log rotate

### DISCUSSION THREAD

**Discussion by dec21raj** [martin-helmich/prometheus-nginxlog-exporter#197] [Q&A] [answered]
Hi Team,

First of all Thank you very much for the software.

I installed RPM of this software on CentOS 7.x which works great.
Few of my server are running with "Scientific Linux release 6.7 (Carbon)" where systemd is not supported. So I tried to use Docker version of this software.

Issue I am noticing, when access.log file rotates, exporter stop reading the contents from access.log
Am I running docker instance in a wrong way ?

docker run -d  \
    --name nginx-exporter \
    -v /var/log/nginx/access.log:/var/log/nginx/access.log \
    -v /etc/prometheus/prometheus-nginxlog-exporter.hcl:/etc/prometheus/prometheus-nginxlog-exporter.hcl \
    -p 4040:4040 \
    quay.io/martinhelmich/prometheus-nginxlog-exporter \
    -config-file /etc/prometheus/prometheus-nginxlog-exporter.hcl

**Comment by martin-helmich** [comment, marked as answer]
In general, this should not happen, and I've used this exporter with log rotation successfully. I this this might possibly be related as to _how_ exactly the access log file is rotated. Let me do some digging and get back to you on this. ⏳

**Reply by dec21raj** [reply]
Thanks Martin. 
Please let me know if you need additional information. 

NGINX Logrotate config pasted blow.

 cat /etc/logrotate.d/nginx
/var/log/nginx/*.log {
        daily
        missingok
        rotate 52
        compress
        delaycompress
        notifempty
        create 640 nginx adm
        sharedscripts
        postrotate
                if [ -f /var/run/nginx.pid ]; then
                        kill -USR1 `cat /var/run/nginx.pid`
                fi
        endscript
}

**Reply by martin-helmich** [reply]
You could try using the `copytruncate` option.

**Reply by dec21raj** [reply]
Thanks Martin. That helped.
Looks like its limitation of docker, which can't load the new log file created where inode got changed after rotation.
